src/module/apps/compendium-browser/index.ts

```diff
@@ -331,14 +478,82 @@ class CompendiumBrowser extends Application {
+                            delete this.packLoader.sources.allowed[key]; // just to make sure we clean up
+                            continue;
                         }
+                        source.load = formData.has(`source-${key}`);
                     }
+                    // Make it sorted, so it can be displayed sorted
+                    this.packLoader.sources.allowed = Object.fromEntries(
+                        Object.entries(this.packLoader.sources.allowed).sort()
+                    );
+                    this.packLoader.sources.allowEmptySources = formData.has("allow-empty-sources");
+                    this.packLoader.sources.allowUnknownSources = formData.has("allow-unknown-sources");
+                    await game.settings.set("pf2e", "compendiumBrowserSources", this.packLoader.sources);
+
+                    await this.#resetInitializedTabs();
                     this.render(true);
                 });
+
+                const sourceSearch = form.querySelector("input[data-element='setting-sources-search']");
+                sourceSearch?.addEventListener("input", (event) => {
+                    const value = (event.currentTarget as HTMLInputElement)?.value?.trim();
+                    const allElements = form.querySelectorAll("label[data-element='setting-source']");
+                    const toggleElement = form.querySelector(
+                        "input[data-action='setting-sources-toggle-visible']"
+                    ) as HTMLInputElement;
+
+                    // first reset / show elements
+                    for (const element of allElements) {
+                        element.classList.remove("hidden");
+                    }
+
+                    if (toggleElement) {
+                        toggleElement.checked = false;
+                    }
+
+                    if (!value) {
+                        return;
+                    }
+
+                    // hide everything not found
+                    const elements = form.querySelectorAll(
+                        `label[data-element='setting-source']:not([data-name*='${value}' i])`
+                    );
+                    for (const element of elements) {
+                        element.classList.add("hidden");
+                    }
+                });
+
+                const toggleElement = form.querySelector("input[data-action='setting-sources-toggle-visible']");
+                toggleElement?.addEventListener("click", (event) => {
+                    const value = (event.currentTarget as HTMLInputElement)?.checked;
+                    const allElements = form.querySelectorAll(
+                        "label[data-element='setting-source']:not(.hidden) input[type='checkbox']"
+                    );
+                    for (const element of allElements as NodeListOf<HTMLInputElement>) {
+                        element.checked = value;
+                    }
+                });
```

Comment:
This needs to be improved but I'm running out of time right now. Some pointers:
- `toggleElement` should be at the top so it can be reused in the `sourceSearch` listener instead of redefining it.
- Remove the single quotes from the selectors, they are not needed.
- Use `querySelector<HTMLInputElement>` (or better `htmlQuery<HTMLInputElement>`) instead of asserting the type.
- Try to avoid using `not` in the css selectors. The whole thing can probably be done in a single loop by checking `element.classList.has("hidden")` and checking `element.dataset.name` for the search value.

I will look at the rest of the PR in a few hours.

Reply:
All of this makes sense, will work on this. I think I was kind of tunnel visioned after looking into the css selector for the checking if the data element has the value.